### PR TITLE
Remove jQuery from Graph.js

### PR DESCRIPTION
**Pull request by Fullchee**
List of changes
- $('#id') to document.getElementByID()
- getURLParameters() to URL.searchParams
- $.ajax() to fetch()
  - I added text to the fetch() error from the headers.
  - That should help with errors like if the URL was typed incorrectly
- $.extend() to JS Spread operators for arrays

### CODE REVIEW

File: js/components/graph/Graph.js [hunk 1]
```diff
@@ -87,14 +88,29 @@ export default class Graph extends React.Component {
                 graphName = localStorage.getItem('active-graph') || 'Computer Science';
             }
         }
-
+
```
Comment (david-yz-liu) [inline js/components/graph/Graph.js]: Remove the whitespace here.

File: js/components/graph/Graph.js [hunk 2]
```diff
@@ -87,14 +88,29 @@ export default class Graph extends React.Component {
...
+        
         graphName = graphName.replace('-', ' ');
 
-        $.ajax({
-            dataType: 'json',
-            url: 'get-json-data',
-            data: { 'graphName': graphName },
-            success: function (data) {
+        let url = new URL('/get-json-data', document.location);
+        console.log(url);
```
Comment (david-yz-liu) [inline js/components/graph/Graph.js]: Remove this line.

File: js/components/graph/Graph.js [hunk 3]
```diff
@@ -137,12 +153,12 @@ export default class Graph extends React.Component {
...
                 });
-            }.bind(this),
-            error: function (xhr, status, err) {
-                console.error('graph-json', status, err.toString());
-            }
-        });
-
+            })
+            .catch((err) => {
+                console.log("Fetch API failed.");
```
Comment (david-yz-liu) [inline js/components/graph/Graph.js]: You can make each of these a `console.error`.

**Comment by Fullchee** [comment]
I've updated the PR according to your comments.

**Pull request merged**